fendermint/eth/api/src/state.rs

```diff
@@ -395,6 +407,30 @@ where
             Ok(Some(data))
         }
     }
+
+    pub async fn get_actor_type(
+        &self,
+        address: &et::H160,
+        height: FvmQueryHeight,
+    ) -> JsonRpcResult<ActorType> {
+        let addr = to_fvm_address(*address);
+        let Some((
+            _,
+            ActorState {
+                code: actor_type_cid,
+                ..
+            },
+        )) = self.client.actor_state(&addr, height).await?.value
+        else {
+            return Ok(ActorType::Inexistent);
+        };
+        let registry = self.client.builtin_actors(height).await?.value.registry;
+        let ret = match registry.into_iter().find(|(_, cid)| cid == &actor_type_cid) {
```

Comment:
Have you considered what I commented about caching these lookups?

Reply:
Added a [TODO in Linear](https://linear.app/interplanetary-consensus/issue/ENG-616/caching-the-result-of-get-actor-type) for that.